api/log.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	chimiddleware "github.com/go-chi/chi/middleware"
+	gcontext "github.com/netlify/gocommerce/context"
+)
+
+func newStructuredLogger(logger *logrus.Logger) func(next http.Handler) http.Handler {
+	return chimiddleware.RequestLogger(&structuredLogger{logger})
+}
+
+type structuredLogger struct {
+	Logger *logrus.Logger
+}
+
+func (l *structuredLogger) NewLogEntry(r *http.Request) chimiddleware.LogEntry {
+	entry := &structuredLoggerEntry{Logger: logrus.NewEntry(l.Logger)}
+	logFields := logrus.Fields{}
+
+	logFields["component"] = "api"
+	logFields["method"] = r.Method
+	logFields["path"] = r.URL.Path
+
+	if reqID := gcontext.GetRequestID(r.Context()); reqID != "" {
```

Comment:
That is _per logline_. I mean as a way to track across the scope of the request. 

https://github.com/netlify/proxy/blob/master/server/helpers.go#L40